nmostesting/suites/IS1101Test.py

```diff
@@ -13,14 +13,344 @@
+        for output in self.outputs:
+            _, response = TestHelper.do_request(
+                "GET", self.compat_url + "outputs/" + output + "properties/"
+            )
+            if response.status_code == 200:
+                outputs_properties_json = []
+                outputs_properties_json.append(response.json())
+                for output in outputs_properties_json:
+                    if output["connected"]:
+                        self.connected_outputs.append(output["id"])
+            else:
+                return test.FAIL(response.json())
+        if len(self.connected_outputs) == 0:
+            return test.UNCLEAR("None Outputs support edid.")
+        return test.PASS()
+
+    def test_03_03(self, test):
+        """
+        Verify that all connected outputs do not have
+        a signal as test 0 put all of the receivers inactive.
+        """
+        if len(self.connected_outputs) == 0:
+            return test.UNCLEAR("no connected outputs")
+        for output_id in self.connected_outputs:
+            _, response = TestHelper.do_request(
+                "GET", self.compat_url + "outputs/" + output_id + "/properties/"
+            )
+            if response.status_code == 200:
+                if response.json()["status"]["state"] == "signal_present":
+                    self.active_connected_outputs.append(response.json())
+            else:
+                return test.FAIL(response.json())
+        if len(self.active_connected_outputs) != 0:
+            return test.UNCLEAR(
+                "Connected output have a signal while all receivers are inactive"
+            )
+        return test.PASS()
+
+    def test_03_04(self, test):
+        """
+        Verify that connected outputs supporting EDID behave according to the RAML file.
+        """
+        if len(self.connected_outputs) == 0:
+            return test.UNCLEAR("no connected outputs")
+        for output_id in self.connected_outputs:
+            _, response = TestHelper.do_request(
+                "GET", self.compat_url + "outputs/" + output_id + "/properties/"
+            )
+            if response.status_code == 200:
+                if response.json()["edid_support"]:
+                    self.edid_connected_outputs.append(response.json()["id"])
+            else:
+                return test.FAIL(response.json())
+        if self.edid_connected_outputs == 0:
+            return test.UNCLEAR("Outputs not supporting edid")
+        return test.PASS()
+
+    def test_03_04_01(self, test):
+        """
+        Verify that an output indicating EDID support behaves according to the RAML file.
+        """
+        if len(self.edid_connected_outputs) == 0:
+            return test.UNCLEAR("no edid connected outputs")
+        for output_id in self.edid_connected_outputs:
+            _, response = TestHelper.do_request(
+                "GET", self.compat_url + "outputs/" + output_id
```

Comment:
Why only make this request for these outputs not all outputs?

**Edit:** I see there is `test_03_05` as well... but there is no difference in the sub-endpoints array response in these two cases, so is there any reason to separate these? And still, what about unconnected Outputs?

And then again, what is being tested that isn't already tested by the `auto_` tests?

Reply:
This test is about "connected" outputs. It is not possible to test EDID for outputs that are not connected because the EDID comes form the connected device. This is in the test plan. A new version v0.7 of the test plan will indicate a test 3.6 for "disconnected" outputs if there is something to test about them. Similarly there is in v0.6 a test for "disconnected" inputs test 1.8

These tests are doing exactly what is requested in the test plan. test_03_04_01 performs the first request in the test plan to A?/outputs/{output_id}/ The test plan assumes that automatic testing has verified the RAML compliance (mostly) so this request mainly check that the stored id is still valid.

Reply:
OK. Refactoring if helpful can be addressed later. :-)